pep-0654.rst

```diff
@@ -267,6 +266,69 @@ as a shorthand for matching that type: ``eg.split(T)`` divides ``eg`` into the
 subgroup of leaf exceptions that match the type ``T``, and the subgroup of those
 that do not (using the same check as ``except`` for a match).
 
+Subclassing Exception Groups
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to subclass exception groups, but when doing that it is
+usually necessary to specify how ``subgroup()`` and ``split()`` should
+create new instances for the matching or non-matching part of the partition.
+``BaseExceptionGroup`` exposes an instance method ``derive(self, excs)``
+which is called whenever ``subgroup`` and ``split`` need to create a new
+exception group. The parameter ``excs`` is the sequence of exceptions to
+include in the new group. Since ``derive`` has access to self, it can
+copy data from it to the new object. For example, if we need an exception
+group subclass that has an additional error code field, we can do this:
+
+.. code-block::
+
+    class MyExceptionGroup(ExceptionGroup):
+        def __new__(cls, message, excs, errcode):
+            obj = super().__new__(cls, message, excs)
+            obj.errcode = errcode
+            return obj
+
+        def derive(self, excs):
+            return MyExceptionGroup(self.message, excs, self.errcode)
+
+
+Note that we override ``__new__`` rather than ``__init__``; this is because
+``BaseExceptionGroup`` has a ``__new__`` method whose signature doesn't match
+that of our new type.  Note also that our ``derive`` function does not
```

Comment:
Well, it must override "the constructor" (i.e. either new or init) because it has a different signature; it must override new rather than init because ExceptionGroup has immutable fields (message and exceptions).

Reply:
Is it because of immutability, or because of the selection between ExceptionGroup and BaseExceptionGroup depending on their types? Or both?